sw/device/examples/hello_world/hello_world.c

```diff
@@ -54,17 +55,16 @@ void trap_handler(uint32_t mepc, char c) {
 int main(int argc, char **argv) {
   uart_init(UART_BAUD_RATE);
 
+  base_set_stdout(uart_stdout);
+
   pinmux_init();
   // Enable GPIO: 0-7 and 16 is input, 8-15 is output
   gpio_init(0xFF00);
 
   spid_init();
   // Add DATE and TIME because I keep fooling myself with old versions
-  uart_send_str(
-      "Hello World! "__DATE__
-      " "__TIME__
-      "\r\n");
-  uart_send_str("Watch the LEDs!\r\n");
+  base_printf("Hello World! %s %s\r\n", __DATE__, __TIME__);
```

Comment:
Seems like this will not be portable with DV since we intend to bypass the string processing entirely.

Reply:
Correct me if I'm wrong, but the mechanism DV is planning to use for printing is roughly:
- Write (to some special DV peripheral in the testbench) the pointer to the format string (which I think you're assuming is a constant in .rodata?)
- Write the contents of the stack pointed to by the va_list.
- Evaluate printf offline later.

If so, yeah, I can't think of a way you could support %s. I think the right solution is to note that they won't work correctly, and, when DV post-processing is implemented, make sure that it processes %s and %z as if they worked, but, since you won't have a snapshot of RAM available, you can just print out some error, instead.

Also, I should point out that since Rust doesn't use a va_list mechanism, when we move to Rust, we'll probably have less of a bad time dumping stuff for DV (since the format string is parsed by the Rust compiler, not the runtime printing function).

Reply:
You should remove the `%s` use from the `hello_world` example so that when DV goes to run the most basic example it doesn't break.